framework/main.py

```diff
@@ -10,77 +10,122 @@
 import sys
 
 
-def init_argparser():
+def init_argparser(args=None):
     """
     Initialization  of argument parse library with it's arguments
     Args:
-        None
+        args: list of args to parse in `sys.argv` form. Useful for testing
+            passed to `ArgumentParser.parse_args` . default to None (use `sys.argv`)
 
     Returns:
         object: Parsed arguments - returned from parser.parse_args()
     """
+    server_args_parser = ArgumentParser(add_help=False)
+
+    # sharing arguments using parent parsers.
+    # https://stackoverflow.com/a/56595689 and https://stackoverflow.com/a/7498853
 
-    parser = ArgumentParser()
-    parser.add_argument(
+    # TODO These really should be config file things,
+    server_args_parser.add_argument(
+        "-s",
         "--server-url",
         help="The URL of the Resource Locker Server",
         required=True,
         action="store",
     )
-    parser.add_argument(
+    server_args_parser.add_argument(
+        "-t",
         "--token",
         help="Token of the user that creates API calls",
         required=True,
         action="store",
     )
-    parser.add_argument(
-        "--release", help="Use this argument to release a resource", action="store_true"
+    server_args_parser.add_argument(
+        "--resume-on-connection-error",
+        help="Use this argument in case you don't want to break queue execution"
+        " in the middle of waiting for queue status being FINISHED",
+        action="store_true",
     )
-    parser.add_argument(
-        "--lock", help="Use this argument to lock a resource", action="store_true"
+
+    # TODO Come up with a better way of handling defaults in ResourceLocker
+    # Doing interval here since it is required as part of `--resume-on-connection-error`
+    interval_default = 15
+    server_args_parser.add_argument(
+        "-i",
+        "--interval",
+        help=f"How many seconds to wait between each call (default {interval_default})",
+        type=int,
+        action="store",
+        default=interval_default,
     )
-    parser.add_argument(
-        "--check", help="Use this to check if a resource is available", action="store_true"
+
+    parser = ArgumentParser(parents=[server_args_parser])
+
+    # Parent of all sub commands that take a search string
+    search_parser = ArgumentParser(add_help=False)
+    search_parser.add_argument(
+        "search_string",
+        metavar="SEARCH",
+        help="Specify the label or the name of the lockable resource",
     )
-    parser.add_argument(
-        "--resume-on-connection-error", help="Use this argument in case you don't want to break queue execution"
-                                             " in the middle of waiting for queue status being FINISHED", action="store_true"
+
+    # Parent of all sub commands that take a signoff string
```

Comment:
this comment is doubled and both times in the incorrect place, i think you wanted to say that the `parser` is the parent of all subcommands, but the comment is located at the `search_parser` and at the `signoff_parser` instead.